src/Url.php

```diff
@@ -1066,4 +1066,33 @@ public static function peelUrl($haystack, $prelude_string, $wrapper_start, $wrap
     }
     return FALSE;
   }
+
+  /**
+   * Checks if given URL matches a list of candidates for a default document.
+   *
+   * @param string $url
+   *   The URL to be tested
+   * @param array $candidates
+   *   A list of potential document names that could be indexes.
+   *   Defaults to "default" and "index".
+   *
+   * @return mixed
+   *   string - The base path if a matching document is found.
+   *   bool - FALSE if no matching document is found.
+   */
+  public static function getRedirectIfIndex($url, $candidates = array("default", "index")) {
+    // Assume no match
+    $root_path = FALSE;
+
+    // Filter through parse_url to separate out querystrings and etc.
+    $parsed_url = parse_url($url, PHP_URL_PATH);
+    $filename = strtolower(pathinfo($parsed_url, PATHINFO_FILENAME));
+
+    // Test parsed URL
+    if (!empty($filename) && in_array($filename, $candidates)) {
+      $root_path = substr($parsed_url, strlen($parsed_url) - strlen($candidate));
```

Comment:
Needs to be reassembled with $query.

Otherwise we will get redirects lost  because the paths for 
http://wireless.fcc.gov/index.htm?job=home
and 
http://wireless.fcc.gov/index.htm?job=forms_and_fees
would both reduce to 
redirect-wireless/   and I can't redirect that to both nodes migrated in.   The first one migrated would squat on the redirect.... and it would be wrong.